### ISSUE TITLE
Window with extendedclientarea incorrectly sized on fullscreen

### ISSUE THREAD

**Issue by giacarrea** [AvaloniaUI/Avalonia#17497]
### Describe the bug

I have a simple window with a customized titlebar. 

When not maximized, the elements are fully displayed.
![image](https://github.com/user-attachments/assets/c6cc2384-5820-4faf-95e6-411207641c6d)

When maximized, the window seem larger than the screen.
![image](https://github.com/user-attachments/assets/8e77e74b-04c5-49fd-9ded-572f1240f9f7)

This behavior can be mitigated by adding a maximized specific styling: 
`    <Window.Styles>
        <Style Selector="Window[WindowState=Maximized]">
            <Setter Property="Padding" Value="8" />
        </Style>
    </Window.Styles>`
    
Here's a solution embedding the issue, derived from the avalonia sample repository:    
[schlorping.zip](https://github.com/user-attachments/files/17735987/schlorping.zip)

### To Reproduce

Compare the elements' edges when the window is maximized VS when it's not. (seem to only occur on windows)

### Avalonia version

11.2.0

### OS

Windows

**Comment by mgarstenauer** [comment]
Instead of using a `Style` with a hardcoded `Padding`, you can use:

```xml
<Panel Padding="{Binding $parent[Window].OffScreenMargin}">
    ...
</Panel>
```

See also:
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/MainWindow.xaml#L59

Maybe this should be built into the default `Window` control theme. I think that could solve the problem for everyone – but it would mess up the style for users that already apply `OffScreenMargin` to their content.

**Comment by pavelovcharov** [comment]
It looks like the OffScreenMargin is not calculated correctly. The MainView doesn't fit the screen (tested on Windows 11)

![image](https://github.com/user-attachments/assets/05163fae-f777-4e88-944d-6243a69efef1)

**Comment by mgarstenauer** [comment]
@pavelovcharov The screenshot looks fine. The custom title "Content in Title Bar" is fully visible. I don't see a problem.

Let me explain:

- I assume you have enabled "Extend Client Area to Decorations". That means the client area also includes the window border (+1px on all sides).
- The desktop minus the taskbar is 2560×1392.
- The window is maximized. In this case Windows hides the window border.  1 px of the window is outside the visible area. That's how you end up with `MainView.Bounds.Width == 2562`.

If you select "MainWindow" in DevTools and look for `OffScreenMargin` you should see a value like `7,7,7,7`.

(Tested with Avalonia 11.1.3)

**Comment by pavelovcharov** [comment]
@mgarstenauer Yes, I have enabled the ExtendClientAreaToDecorationsHint option. 
The MainView is the user content. Its width is 2562px while the screen width is 2560px. As a result, the user content is not completely visible on the screen.

**Comment by pavelovcharov** [comment]
Or should we add this 1pixel margin manually?

**Comment by mgarstenauer** [comment]
I am not sure how the 1 pixel margin is supposed to be handled – or whether it is actually a bug. It is discussed here as well: #17202

I think the authors of the Win32 window implementation need to clarify the issue.
